pkg/grpc/service/study-management_endpoints.go

```diff
@@ -143,6 +143,10 @@ func (s *studyServiceServer) SaveSurveyToStudy(ctx context.Context, req *api.Add
 		}
 		newSurvey.VersionID = utils.GenerateSurveyVersionID(surveyHistory)
 	}
+
+	if newSurvey.Published == 0 {
```

Comment:
For current behaviour, always overriding the value with current time, would be safer, right? 
Later, if we decide to support custom published dates, we can change.

Reply:
Yeah, if we are not using it for specific reason, I would completely remove it from the API (the request should ignore the published value passed) but this might be a breaking change if it was present, so we need to consider why it was added in the first place and if it is reasonably safe to remove.

Reply:
In the API you simply push a survey object to the server. That data type has all the attribute a survey has, so in theory, you could set everything. But this attribute, even if set from the outside would be overwritten by the server, to have the "real" published date tracked, which would actually make sense. 
I can imagine that sometimes in the future, we could use "published" value manually defined for the purpose of scheduling when an update becomes available, but have no use-case now, that needs this. For having a "published" value that is smaller than timestamp of now, the only reason I can imagine is to fix some past mistake / data inconsistency in the system - which ideally should not happen in the first place.

Reply:
I think, previously the python client handled the logic of setting the published date. That script need to be changed anyway to be able to work with the new history model slightly (it becomes simpler).

Reply:
I see, the db model is leaking into the API. I'm not a fan of this because now two very different layers are tied up and changes in the db data model result in a breaking change in the API layer and also makes everything confusing. We should completely avoid doing this from now on but this is outside of the scope of this review.

Reply:
I see what you mean, but here not completely - we have the possibility to change data type between DB and API layer - in this particular case it's only so, because we use the same data type for uploading the survey as for downloading a survey definition. When downloading infos from the backend, the published attribute might be useful for multiple reasons, so it's good that it's exposed. Just with grpc I find it quite cumbersome to define slight variations of a type, so if possible I reuse types, even if sometimes one or the other attribute might be irrelevant. (Not always, depends a little bit on the complexity of the type and semantics...)
So this particular "inconsistency" comes because we did not want to define separate types for uploading and downloading surveys.